autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
 
     def __str__(self):
+        """
+        Operator overloading to return the string representation of the class objects
+
+        Returns
+        -------
+        str
+            String representation of query sequence ID, followed by total number of hits and finally the highest
+            bit score of all the hits
+        """
         return f"{self.qseqid}; {len(self.sseqids)} sseqids; top hit by bitscore: {self.get_top_hit()}"
 
     def __eq__(self, other_hit):
+        """
+        Operator overloading to compare two objects of the class
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            Other sseqid hit dictionary to compare with
+
+        Returns
+        -------
+        Boolean
+            True if both dcitionaries are equal else False
+        """
         if self.qseqid == other_hit.qseqid:
             return True
         else:
             return False
 
     def __add__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
+        """
+        Operator overloading to update (add) the sseqids dictionary with the other sseqid hit dictionary
+        for a specific query sequence
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            {
+            sseqid: {
+                "pident": float(pident),
+                "length": int(length),
+                "mismatch": int(mismatch),
+                "gapopen": int(gapopen),
+                "qstart": int(qstart),
+                "qend": int(qend),
+                "sstart": int(sstart),
+                "send": int(send),
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
+            
+        """
+        assert self == other_hit, f"qseqids do not match! {self} & {other_hit}"
         self.sseqids.update(other_hit.sseqids)
         return self
 
     def __sub__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
-        self.sseqids.pop(other_hit.sseqid)
+        """
+        Operator overloading to remove (subtract) the other sseqid hits dictionary from the sseqids dictionary
+        for a specific query sequence
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            {
+            sseqid: {
+                "pident": float(pident),
+                "length": int(length),
+                "mismatch": int(mismatch),
+                "gapopen": int(gapopen),
+                "qstart": int(qstart),
+                "qend": int(qend),
+                "sstart": int(sstart),
+                "send": int(send),
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
```

Comment:
Same comments as above. `other_hit` is not a `dict`, a `dict` is not returned and the parameter string is wrong